src/java.base/share/classes/jdk/internal/classfile/impl/DirectCodeBuilder.java

```diff
@@ -617,20 +618,24 @@ private void writeParsedLongLabel(int jumpOrInstructionPc, Label nullOrTarget) {
     }
 
     public void writeLookupSwitch(Label defaultTarget, List<SwitchCase> cases) {
+        cases = new ArrayList<>(cases); // cases may be untrusted
+        for (var each : cases) {
+            Objects.requireNonNull(each); // single null case may exist
+        }
+        cases.sort(new Comparator<>() {
+            @Override
+            public int compare(SwitchCase c1, SwitchCase c2) {
+                return Integer.compare(c1.caseValue(), c2.caseValue());
```

Comment:
isn this an implicit null check?

Reply:
Yes, but this doesn't run when the list has just one element of `null` - see case https://github.com/openjdk/jdk/pull/26229/files#diff-cb987200ae23ff7522bd692d4707a012b115bf778a1265ba11db0b3bf5c9aa7cR98